### ISSUE TITLE
Dependency Updates for internal Repositories not working for all repository managers

### ISSUE THREAD

**Issue by sischi** [DependencyTrack/dependency-track#2075]
Hi there,

I was waiting for the new DependencyTrack version 4.6 because it supports repository authentication for internal repositories, so that checking versions of internal components to be the latest will work for us.

So I configured our internal repository with authentication via frontend, uploaded a bom that uses an internal dependency and ... no version update information is displayed :(

After checking the logs and browsing the relevant code of dependency-track I was able to find the reason, that no version update is shown. It turns out that the dependency-track checks the version of a component against the `<latest>` information found in the `maven-metadata.xml` from the remote repository, as in the following lines of code:

https://github.com/DependencyTrack/dependency-track/blob/7878b039b7b61e5aa3b9a1af9ddab7de738f2925/src/main/java/org/dependencytrack/tasks/repositories/RepositoryMetaAnalyzerTask.java#L96-L111

The problem now is, that the `maven-metadata.xml` in our internal repository contains no `<latest>`  information. I have then started some research about the maven metadata and those information and it seems like maven is not responsible for updating/setting the `<latest>` information in such `maven-metadata.xml`, except the deployed artifact is a **maven plugin**. I couldn't find any "official" resources, but it was explained [here (Nexus Repository)]() and [here (stackoverflow)](https://stackoverflow.com/a/47704838). I can reproduce this situation in my local repository, where the `<latest>` information is contained in the `maven-metadata-local.xml` for maven plugin artifacts but not for other maven artifacts. So the `<latest>` information for non maven plugin artifacts seem to be updated by the repository manager and the one we use (gitlab) seems not to support this.

What gets updated for both, maven plugin artifacts and non maven plugin artifacts, is the `<release>` information in the `maven-metadata.xml`. Like said in the [maven documentation](https://maven.apache.org/ref/3.8.6/maven-repository-metadata/repository-metadata.html) the `<latest>` contains *"the last version added to the directory, including both releases and snapshots"*, whereas the `<release>` contains *"the last version added to the directory, for the releases only"*.

This said, wouldn't it be better to compare a components version with the `<release>` instead of `<latest>` information in general? At least I would love to see a workaround for those repositories where the `maven-metadata.xml` **does not** contain `<latest>` but **does** contain `<release>` information. 

Thanks in advance!

**Comment by syalioune** [comment]
It certainly make sense to be flexible about it.
Looking at different implementations, I get what you described :

## Gitlab maven registry

```xml
<metadata>
<groupId>redacted</groupId>
<artifactId>redacted</artifactId>
<versioning>
<release>0.0.1</release>
<versions>
<version>0.0.1</version>
</versions>
<lastUpdated>20221025075411</lastUpdated>
</versioning>
</metadata>
```

## Nexus registry

```xml
<metadata modelVersion="1.1.0">
<groupId>redacted</groupId>
<artifactId>redacted</artifactId>
<versioning>
<latest>1.0.0</latest>
<release>1.0.0</release>
<versions>
<version>1.0.0</version>
</versions>
<lastUpdated>20210805201810</lastUpdated>
</versioning>
</metadata>
```

DT should preferably check for `release` tag existence and if not present use `latest` tag in https://github.com/DependencyTrack/dependency-track/blob/64a8e0b9acc3d3d69d8ce3bc358bed93797aa802/src/main/java/org/dependencytrack/tasks/repositories/MavenMetaAnalyzer.java#L101

**Comment by msymons** [comment]
As a user of Nexus Repository, I would love to see a solution to this Maven problem.  I ran into it when testing repository authenticaton when it was first enabled in 4.6 (SNAPSHOT) back in June 2022. 

The authentication worked great...  but when all my up-to-date internal components were then reported as being out-of-date by Dependency-Track (due to snapshots) I decided to delete the internal repo definition from DT.  

(I asked about the problem on Slack but could not create an issue here until 4.6.0 was released).

**Comment by stevespringett** [comment]
@msymons sounds like the Nexus repo may be setup in "mixed" mode, which is going to make things more difficult.
https://help.sonatype.com/repomanager3/nexus-repository-administration/formats/maven-repositories

Most repos I've seen have been setup to either accept releases or snapshots, but not both (mixed).

**Comment by msymons** [comment]
Our internal releases are in an NXRM repository X of type "hosted" and version policy "release"
Our internal snapshots are are in an NXRM repository Y of type "hosted" and version policy "snapshot"

The actual "download" NXRM repo Z is of type "group" and version policy "release". Z has X and Y as members, along with proxied repos such as maven central, etc.

Thus,  matching comment of @syalioune,  our internal components have `maven-metadata.xml` that contain (say):

```
<latest>1.1.1-SNAPSHOT</latest>
<release>1.1.0</release>
```

If I look instead at release repo X,  `maven-metadata.xml`  does not have a `latest` versioning.  Just `release`.

**Comment by syalioune** [comment]
@sischi I'm curious about the credentials you configured in DT frontend. Since, you're using Gitlab which doesn't provide a basic auth capability to its rest api (to my knowledge) and DT only perform basic auth, it should not work anyway

**Comment by sischi** [comment]
@syalioune Yes, I did some research about that when I saw that DT will support basic authentication for private repositories. I currently have no reference but I found, that Basic Authentication will work if you simply use the token generated in gitlab as a basic auth password and I can confirm that it works.

**Comment by msymons** [comment]
I have tested the fix of @syalioune in the latest 4.7 SNAPSHOT and things are now working great for me.  In fact, better than great,  I did not realise just how useful it would be to see "Latest Version" information for internal components until it was actually working!

So, from my perspective, this issue can be closed as "completed in 4.7".  As long as @sischi is happy.

**Comment by sischi** [comment]
Thanks for your support @syalioune and @msymons! I have not tested it yet, but your implementation looks great, @syalioune.

Yes @msymons I'm very happy now 😊 so I will close this issue. Good job 👌
